server/manager/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java

```diff
@@ -214,14 +214,8 @@ public void run() {
           TableId tableId = tls.extent.tableId();
           TableConfiguration tableConf = this.master.getContext().getTableConfiguration(tableId);
 
-          MergeStats mergeStats = mergeStatsCache.get(tableId);
-          if (mergeStats == null) {
-            mergeStats = currentMerges.get(tableId);
-            if (mergeStats == null) {
-              mergeStats = new MergeStats(new MergeInfo());
-            }
-            mergeStatsCache.put(tableId, mergeStats);
-          }
+          MergeStats mergeStats = mergeStatsCache.computeIfAbsent(tableId,
+              k -> currentMerges.getOrDefault(k, new MergeStats(new MergeInfo())));
```

Comment:
This always constructs the the 'default' `new MergeStats(new MergeInfo()))`, even if already exists in `currentMerges`. To avoid that, you could use the following instead of `getOrDefault`:

```suggestion
              k -> {
                var mergeStats = currentMerges.get(k);
                return mergeStats != null ? mergeStats : new MergeStats(new MergeInfo());
              });
```

Reply:
Now you got me thinking, how does ```map.getOrDefault(k, new MergeInfo())``` behave differently from ```log.debug("Merge: {}", new MergeInfo())```?  Does slf4j have a special ability to not call the method on the interface based on logging level vs the default method on the Map interface, which will always execute the method?

Reply:
I guess I am confusing how sl4j doesn't create _unnecessary_ objects with the creation of an object when it is passed it as a parameter.  If ```new Object()``` is passed as a parameter to a method, it will always create the object when the method is executed.

Reply:
To clarify, this code will throw an exception even at the "info" level:
```
public static void main(String[] args) {
        log.info("Start main");
        log.debug("debug called: {}", expensiveCall());
        log.info("Finished");
    }
    public static String expensiveCall() {
        if (log.isInfoEnabled())
            throw new UnsupportedOperationException("Do not call");
        return "expensive string";
    }
```

Reply:
With log4j 2 - it appears that lambdas can be used to provide lazy behavior.
`log.debug("debug called: {}", () -> expensiveCall());`
Would avoid the call if debug level logging was not enabled.

Reply:
This does work if you explicitly call the Log4j logger.  Lambdas aren't implemented yet in slf4fj.  There is an alpha version though: https://jira.qos.ch/browse/SLF4J-371

Reply:
Yeah, the same issue would apply to our slf4j debug statments... which is why we typically wrap stuff in an `if` block to check the log level `.isTraceEnabled()` first, to avoid making expensive objects.